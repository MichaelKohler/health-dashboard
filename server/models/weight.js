'use strict';

module.exports = (sequelize, DataTypes) => {
  const Weight = sequelize.define('Weight', {
    weight: DataTypes.FLOAT,
  }, {});

  return Weight;
};
