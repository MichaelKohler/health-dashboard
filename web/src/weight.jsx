import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Fab from '@material-ui/core/Fab';
import CircularProgress from '@material-ui/core/CircularProgress';
import AddIcon from '@material-ui/icons/Add';

import history from './history';
import Table from './weight-table.jsx';

const useStyles = makeStyles((theme) => ({
  appBarSpacer: theme.mixins.toolbar,
  container: {
    paddingTop: theme.spacing(4),
    paddingBottom: theme.spacing(4),
  },
  paper: {
    padding: theme.spacing(2),
    display: 'flex',
    overflow: 'auto',
    flexDirection: 'column',
  },
  fab: {
    position: 'absolute',
    bottom: theme.spacing(2),
    right: theme.spacing(2),
  },
}));

const mapStateToProps = (state) => ({
  weights: state.weights,
  isFetchingHealth: state.isFetchingHealth,
  failedFetchingHealth: state.failedFetchingHealth,
});

export function Weight(props) {
  const classes = useStyles();

  return (
      <section>
          <div className={ classes.appBarSpacer }/>
          <Container maxWidth="lg" className={ classes.container }>
              <Grid container spacing={ 3 }>
                  <Grid item xs={ 12 } md={ 12 } lg={ 12 }>
                      { props.isFetchingHealth && (
                      <Paper className={ classes.paper }>
                          <p>Fetching...</p>
                          <CircularProgress size={ 15 }/>
                      </Paper>
                      ) }
                      { props.weights.length > 0 && (
                      <Table rows={ props.weights }/>
                      ) }
                      { props.failedFetchingHealth && (
                      <Paper className={ classes.paper }>
                          <p>Oh no, something went wrong!</p>
                      </Paper>
                      ) }
                  </Grid>
              </Grid>
              <Fab aria-label="Add" className={classes.fab} color="primary" onClick={ () => history.push('/weight/add')}>
                  <AddIcon/>
              </Fab>
          </Container>
      </section>
  );
}

Weight.propTypes = {
  weights: PropTypes.array,
  isFetchingHealth: PropTypes.bool,
  failedFetchingHealth: PropTypes.bool,
};

export default connect(
  mapStateToProps,
)(Weight);
