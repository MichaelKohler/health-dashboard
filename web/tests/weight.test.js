/* global it, expect */
/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Weight } from '../src/weight.jsx';

it('renders correctly', () => {
  const data = [{
    createdAt: '2019-08-07T08:00:03Z',
    weight: 80,
  }];

  const tree = renderer
    .create(<MemoryRouter><Weight weights={ data }/></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/* eslint-enable react/jsx-filename-extension */
