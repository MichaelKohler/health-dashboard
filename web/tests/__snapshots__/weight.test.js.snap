// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <div
    className="makeStyles-appBarSpacer-1"
  />
  <div
    className="MuiContainer-root makeStyles-container-2 MuiContainer-maxWidthLg"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      WeightChart
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
      >
        WeightTable
      </div>
    </div>
    <button
      aria-label="Add"
      className="MuiButtonBase-root MuiFab-root makeStyles-fab-4 MuiFab-primary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
    </button>
  </div>
</section>
`;
